extensions/32-bit-address-space/README.md

```diff
@@ -0,0 +1,58 @@
+# General Design
+
+**Extension State: Under Development**
+**Requires: CP1.14**
+**CP Bit: CP1.16**
+
+# Overview
+
+This extension adds a new concept of _modes_ to the ISA. Modes are used to control things about the processor
+state which are binary-incompatible - that is, binary code written for a mode other than the mode the
+processor is actually in will almost certainly behave incorrectly.
+
+In particular, this extension focuses on the _Real 32-bit address mode_.
+
+# Added Instructions
+
+None!
+
+# Modes
+
+CR `1110`, equivalently `cr14` or "the MODE control register," holds a value indicating the current
+_mode_ of the processor.
+
+## Bare Mode
+
+The mode described by [the base isa](../../base-isa.md) is known as the
+Bare mode (Real mode?). In that mode, pointers are 16 bits and there is no virtual memory, paging,
+memory protection, safe multitasking, or privilege restriction. Bare mode is indicated by a value of
+0 in `cr14`.
+
+## Real 32-bit Address Mode
+
+A new mode known as _Real 32-bit address mode_ is added, indicated by a value of 2 in `cr14`.
+In real 32-bit address mode, all addresses are treated as being 32 bits long. Entering real 32-bit
+address mode _must_ preserve the program counter. Returning to Bare mode from real 32-bit address
+mode must preserve the program counter _if possible_ - if the current program counter interpreted
+as a 16-bit address would not be the address of the current instruction, the behavior of the
+system is undefined.
```

Comment:
Out of curiosity, why the value 2 instead of the value 1? 

```suggestion
A new mode known as _Real 32-bit address mode_ is added, indicated by a value of 2 in `cr14`.
In real 32-bit address mode, all addresses are treated as being 32 bits long. Entering real 32-bit
address mode _must_ preserve the program counter. Returning to Bare mode from real 32-bit address
mode must preserve the program counter _if possible_ - if the current program counter interpreted
as a 16-bit address would not be the address of the current instruction, the behavior of the
system is _unspecified_.
```

Reply:
As per discussion on Discord, bit 2 was used so that bit 1 could stay free to indicate virtual memory. I think this point is settled now?